lib/dom_parser.rb

```diff
@@ -0,0 +1,93 @@
+require_relative 'file_loader.rb'
+
+Tag = Struct.new(:text, :type, :classes, :id, :name, :children, :parent, :closing, :depth)
+
+
+class DomParser
+
+  TYPE_REGEX = /<[a-z]*[0-9]*/
```

Comment:
I like having the Regex as a constant at the start of the file. Also seems good to have multiple regexes rather then a single all encompassing regex. Downside of course is as you discover more complexity in sites that you parse you'll need to add more regex constants to keep up. Also possibly having to change your data structure to accomadate.